plugins/by-name/fidget/default.nix

```diff
@@ -34,650 +28,424 @@ let
+              Keys of the table are each notification group's `key`.
+            '';
+      };
 
-          doneIcon = mkIconOption "✔" "Icon shown when all LSP progress tasks are complete.";
+      lsp = {
+        progress_ringbuf_size = defaultNullOpts.mkNum 0 ''
+          Configure the nvim's LSP progress ring buffer size.
+        '';
 
-          doneStyle = helpers.defaultNullOpts.mkStr "Constant" ''
-            Highlight group for completed LSP tasks.
-          '';
+        log_handler = defaultNullOpts.mkBool false ''
+          Log `$/progress` handler invocations (for debugging).
+        '';
+      };
+    };
 
-          progressTtl = helpers.defaultNullOpts.mkStrLuaFnOr types.ints.unsigned "math.huge" ''
-            How long a message should persist when in progress.
-          '';
+    notification = {
+      poll_rate = defaultNullOpts.mkNum 10 ''
+        How frequently to update and render notifications.
+        Measured in Hertz (frames per second).
+      '';
 
-          progressIcon = mkIconOption ''
-            {
-              pattern = "dots";
-            }
-          '' "Icon shown when LSP progress tasks are in progress";
+      filter = defaultNullOpts.mkLogLevel "info" ''
+        Minimum log level to display.
+      '';
 
-          progressStyle = helpers.defaultNullOpts.mkStr "WarningMsg" ''
-            Highlight group for in-progress LSP tasks.
-          '';
+      history_size = defaultNullOpts.mkNum 128 ''
+        Number of removed messages to retain in history.
 
-          groupStyle = helpers.defaultNullOpts.mkStr "Title" ''
-            Highlight group for group name (LSP server name).
-          '';
+        Set to 0 to keep around history indefinitely (until cleared).
+      '';
 
-          iconStyle = helpers.defaultNullOpts.mkStr "Question" ''
-            Highlight group for group icons.
-          '';
+      override_vim_notify = defaultNullOpts.mkBool false ''
+        Automatically override vim.notify() with Fidget.
+      '';
 
-          priority = helpers.defaultNullOpts.mkUnsignedInt 30 ''
-            Ordering priority for LSP notification group.
-          '';
+      configs =
+        defaultNullOpts.mkAttrsOf (with types; either str notificationConfigType)
+          { default = ''require("fidget.notification").default_config''; }
+          ''
+            How to configure notification groups when instantiated.
 
-          skipHistory = helpers.defaultNullOpts.mkBool true ''
-            Whether progress notifications should be omitted from history.
+            A configuration with the key `"default"` should always be specified, and
+            is used as the fallback for notifications lacking a group key.
           '';
 
-          formatMessage = helpers.defaultNullOpts.mkLua "require('fidget.progress.display').default_format_message" ''
-            How to format a progress message.
-
-            ```lua
-              format_message = function(msg)
```

Comment:
Is the example not helpful anymore? Might be worth keeping in the proper `example` property.